packages/BetterPhpDocParser/Annotation/ChangeResolvedClassInParticularContextForAnnotation.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\BetterPhpDocParser\Annotation;
+
+use Rector\BetterPhpDocParser\PhpDoc\DoctrineAnnotationTagValueNode;
+use Rector\BetterPhpDocParser\ValueObject\ChangeResolvedClassInParticularContextForAnnotationRule;
+use Rector\BetterPhpDocParser\ValueObject\PhpDoc\DoctrineAnnotation\CurlyListNode;
+use Rector\Php80\ValueObject\AnnotationToAttribute;
+
+final class ChangeResolvedClassInParticularContextForAnnotation
+{
+    private const RULES = [
+        [
+            'tag' => 'Doctrine\ORM\Mapping\JoinTable',
+            'value' => 'inverseJoinColumns',
+            'resolved_class' => 'Doctrine\ORM\Mapping\InverseJoinColumns',
+        ],
+    ];
+
+    public function changeResolvedClassIfNeed(
+        AnnotationToAttribute $annotationToAttribute,
+        DoctrineAnnotationTagValueNode $docNode
+    ): void {
+        foreach (self::RULES as $item) {
```

Comment:
Here is still an array type. I'm probably not clear enought, so I'll try to show example in real code:

```hp
final class ChangeResolvedClassInParticularContextForAnnotation
{
    public function __construct()
    {
         // here the Rules is list of specific objects, no array with string anywhere 
        $this->rules = new Rule(...);
    }
}
```

Reply:
done